android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/ImageWithTextFactory.java

```diff
@@ -0,0 +1,75 @@
+package com.vijay.jsonwizard.widgets;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import com.rey.material.widget.ImageView;
+import com.rey.material.widget.TextView;
+import com.vijay.jsonwizard.R;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+import com.vijay.jsonwizard.interfaces.CommonListener;
+import com.vijay.jsonwizard.interfaces.FormWidgetFactory;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ImageWithTextFactory implements FormWidgetFactory {
+
+    private View rootLayout;
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener, boolean popup) throws Exception {
+        List<View> views = new ArrayList<>(1);
+        rootLayout = LayoutInflater.from(context).inflate(getLayout(), null);
+        setWidgetTags(jsonObject);
+        views.add(rootLayout);
+        return views;
+    }
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener) throws Exception {
+        return getViewsFromJson(stepName, context, formFragment, jsonObject, listener, false);
+    }
+
+    private int getLayout() {
+        return R.layout.native_form_text_and_image;
+    }
+
+    private void setWidgetTags(JSONObject jsonObject) throws JSONException {
+        String key = jsonObject.getString(JsonFormConstants.KEY);
+        String openMrsEntityParent = jsonObject.getString(JsonFormConstants.OPENMRS_ENTITY_PARENT);
+        String openMrsEntity = jsonObject.getString(JsonFormConstants.OPENMRS_ENTITY);
+        String openMrsEntityId = jsonObject.getString(JsonFormConstants.OPENMRS_ENTITY_ID);
```

Comment:
Replace `jsonObject.optString(JsonFormConstants.OPENMRS_ENTITY_ID, "");` instead for all the `getString`